skyplane/api/client.py

```diff
@@ -142,3 +142,48 @@ def dataplane(
             return Dataplane(clientid=self.clientid, topology=topo, provisioner=self.provisioner, transfer_config=self.transfer_config)
         else:
             raise NotImplementedError(f"Dataplane type {solver_type} not implemented")
+
+    def download_object(self, bucket_name: str, provider: str, key: str, filename: str):
```

Comment:
Can we move this to another class and add another top-level API primitive? This seem separate from the client itself. You can call it via:
```
obj_store = client.object_store()
```